CohenLabPrinceton/Ventilator-Dev
# Gui integration and checkpoint

the main thing this PR does is integrate the GUI with the rest of the program. hooray!!! It also tries to bring the objects into a uniform system of naming, hooray!!!

For the most part you can skip over the docs and gui changes, as those are either benign or p esoteric stuff for the gui, but feel free to review away. I comment on general changes here and i'll go through and comment on finer-grained changes in the pr

travis & pytest
* add commands to set up xvfb to test gui

docs
* generic cleanup& removal of dead references
* mock pigpio

tests
* cleaned up lots of inconsistent value names, brought in line with using `values.ValueName` in `test_controller` and `test_coordinator`
* added timeouts to potentially infinite tests
* added first round of gui tests! including "run the whole thing!" and "test the small components!" don't judge my lack of tests i'm learning and trying to get there asap!!!

message
* implementing `message.SensorValues` as a carrier class for sensor values. Can't use `enum` values as attribute names, so using the string representation of the names. `to_dict` restores dict of `ValueName: value` pairs. 
* made `AlarmSeverity` enum have explicit numbering so can do comparisons
* gave Alarms unique alarm id number with `itertools.counter`
* added `value` field to Alarm so we know how far the reading went out of bounds.

values
* added `ValueName.IE_RATIO`
* tweaked some defaults, still have yet to implement final values from meeting
* moved `PLOTS` to gui

control_module
* added `PEEP_TIME` to controller -- i think we discussed not having this, but since it was listed in `ValueName` it should have a corresponding entry.
* used `ValueName` rather than strings where applicable.
* `.running` property to get running state

coordinator, process_manager, rpc
* made `get_sensors` return `SensorValues` to be consistent with `control_module`
* made coordinator and process manager exit cleanly -- both processes were hanging
* made `start_process` a single method, rather than being implemented in `__init__` separately
* **edit: now doing the simpler method like this: https://stackoverflow.com/a/1766187 will let @sunnyszy figure out the timeouts** _`rpc_client` a timeout according to https://mail.python.org/pipermail/python-bugs-list/2015-January/260126.html_

gui
* implemented coherent & predictable dark theme, not dependent on system defaults anymore.
* collapsible widgets that don't use the whole cpu
* removed timedplotupdate as it proved not to be a huge benefit and a decent pain in the ass
* made rudimentary alarm manager. very much a placeholder, does not behave like i want it to, but ironing it out to make the logic work and then we can move it to coordinator
* implemented rough draft of alarms with gui - alarms can be triggered by gui and also by controller
* refactored `Vent_Gui.init_ui` to be multiple methods for different sections
* added groupboxes around controls and monitored values to visually offset
* lots and lots and lots of improvements to function of widgets
